File: src/Software Engineer Prepare Kit/03/solution.go
package main

import (
	"regexp"
	"strings"
)

func IsAlphabeticPalindrome(code string) bool {
	var head uint16 = 0
	var tail uint16 = uint16(len(code) - 1)

	regex := regexp.MustCompile(`[^a-zA-Z]`)

	for head < tail {
		for head < tail && regex.MatchString(string(code[head])) {
			head++
		}

		for tail > head && regex.MatchString(string(code[tail])) {
			tail--
		}

		if !strings.EqualFold(string(code[head]), string(code[tail])) {
			return false
		}

		head++
		tail--
	}

	return true
}

File: src/Software Engineer Prepare Kit/03/solution.ts
export default function isAlphabeticPalindrome(code: string): boolean {
	let i = 0
	let j = code.length - 1

	while (i < j) {
		while (i < j && code[i]!.match(/[^a-zA-Z]/)) i++
		while (j > i && code[j]!.match(/[^a-zA-Z]/)) j--

		if (code[i]!.toLowerCase() === code[j]!.toLowerCase()) {
			i++
			j--
		} else {
			return false
		}
	}

	return true
}
